guides/upgrade-mainnet.md

```diff
@@ -0,0 +1,149 @@
+    name: akash
+    server_name: akashd
+    client_name: akashctl
+    version: 0.8.3
+    commit: 11a6e4dfba545073f79732088735f93a5408995c
+    build_tags: netgo,ledger,mainnet
+    go: go version go1.15.5 linux/amd64
+    ```
+    
+2. Set halt-time to ensure your validator node stops at the right time and date: March 08 2021 at 1500 UTC.
+    ```
+   $ perl -i -pe 's/^halt-time =.*/halt-time = 1615195800/' ~/.akashd/config/app.toml
+    ```
+    or you can manually update `app.toml` using `nano` or `vm`
+    
+3. Stop your `akashd` binary or systemd process
+ 
+4. Restart your `akashd` binary or systemd process
+
+## Instructions for upgrade, after the node stops (i.e., after 1500UTC on 8th March, 2021)
+1. Stop your `akashd` binary or systemd process
+2. Check the last block height signed
+
+    Check the last block height signed by your validator:
+```
+$ cat ~/.akashd/data/priv_validator_state.json | jq '.height'
+```
+
+Take the state export from this height (decision on the final height will be coordinated on the mainnet-validator discord channel):
+
+3. Export the `akashnet-1` state
+    
+```
+akashd export --for-zero-height --height <height> > old_genesis_export.json
+```
+4. Fix migration issue by executing following command (This is yet to be fixed on cosmos-sdk: https://github.com/cosmos/cosmos-sdk/issues/8776)
+```sh
+cat old_genesis_export.json | jq '.app_state.auth.accounts= [.app_state.auth.accounts[] | if (.value.public_key!=null and .value.public_key!="") then (.value.public_key.value= (.value.public_key.value | if type=="string" then . else (.threshold = (.threshold| tonumber)) end)) else . end ]' > old_genesis_updated.json
+```
+5. Verify the SHA256 of the (sorted) exported genesis file:
+```
+$ jq -S -c -M '' old_genesis_updated.json | shasum -a 256
+```
+Compare this value with other validators of the network over on #validators-mainnet on the [discord server](https://discord.gg/CuGPqaUW). It's important that everyone can create the same genesis file export.
+
+HASH: [TBD]
+
+6. Install Akash [v0.10.1](https://github.com/ovrclk/akash/releases/tag/v0.10.1)
+
+**Note**: Building the new binary requires Go 1.15+
+```
+git clone https://github.com/ovrclk/akash
+cd akash
+git checkout v0.10.1
+MAINNET=true make install
+```
+
+7. Verify you have the correct version:
+```
+$ akash version --long
+name: akash
+server_name: akash
+version: 0.10.1
+commit: af43b89e47e50bfeedcc35c7ee77229bd258ed0d
+build_tags: netgo,ledger,mainnet
+go: go version go1.15.5 linux/amd64
+build_deps:
+- github.com/99designs/keyring@v1.1.6
+- github.com/ChainSafe/go-schnorrkel@v0.0.0-20200405005733-88cbf1b4c40d
+- github.com/Workiva/go-datastructures@v1.0.52
+....
+```
+
+**Note**: `akashd` and `akashctl` are merged into a single binary, `akash` now.
+
+8.  Migrate exported state from the current v0.8.3 version to the new v0.10.1 version:
+```
+$ akash migrate v0.40 old_genesis_updated.json --chain-id=akashnet-2 > 040_migrated_genesis.json
```

Comment:
@anilCSE Can I add `genesis-time` flag here in this command?

Reply:
May be for unique hash, let's add genesis time as March 8th 1500 UTC

Reply:
Actually we don't need it. We gonna export the state at some height. So block commit time should be same